src/allmydata/util/fileutil.py

```diff
@@ -271,11 +271,13 @@ def write_atomically(target, contents, mode="b"):
     move_into_place(target+".tmp", target)
 
 def write(path, data, mode="wb"):
+    if "b" in mode and isinstance(data, str):
+        data = data.encode("utf-8")
```

Comment:
Seems like this new behavior warrants documentation and/or direct unit test.

Reply:
Done.